### PR TITLE
fix yang connector gnmi reconnect

### PR DESCRIPTION
**Pull request by xvincent**
- Fix yang connector gnmi reconnect:
Move the gRPC channel creation code from class Gnmi ```__init()__``` to ```connect()```

### CODE REVIEW

**Review by ThomasJRyan — CHANGES_REQUESTED** [review]
Seems functional to me. Needs a changelog and likely a untitest to go with it though

File: connector/src/yang/connector/gnmi.py [hunk 1]
```diff
@@ -238,6 +239,29 @@ def __init__(self, *args, **kwargs):
...
         self.metadata = None
+
+    @property
+    def connected(self):
+        """Return True if session is connected."""
+        return self.service
+
+    @property
+    def gnmi(self):
+        """Helper method to keep backwrads compatibility.
```
[the remark this answers is not in the document]
Comment (xvincent) [reply connector/src/yang/connector/gnmi.py]: fixed in 017b3f4

File: connector/src/yang/connector/gnmi.py [hunk 2]
```diff
@@ -226,9 +226,10 @@ class Gnmi(BaseConnection):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.device = kwargs.get('device')
-        dev_args = self.connection_info
-        if dev_args.get('protocol', '') != 'gnmi':
-            msg = 'Invalid protocol {0}'.format(dev_args.get('protocol', ''))
+        self.dev_args = self.connection_info
+        print(f'XAV001: self.dev_args  = {self.dev_args}')
```
[the remark this answers is not in the document]
Comment (xvincent) [reply connector/src/yang/connector/gnmi.py]: removed in a865ce5

File: connector/src/yang/connector/gnmi.py [hunk 3]
```diff
@@ -238,6 +239,29 @@ def __init__(self, *args, **kwargs):
...
+        """
+        return self
+
+    def connect(self):
+        """Connect to device using gNMI and get capabilities.
+
+        Raises:
+            gNMIException: No gNMI capabilities returned by device.
+        """
+        print('XAV002: really connect in def connect()')
```
[the remark this answers is not in the document]
Comment (xvincent) [reply connector/src/yang/connector/gnmi.py]: removed print in a865ce5

**Comment by miott** [comment]
Add a detailed description for this pull request please (and a changelog).

**Comment by xvincent** [comment]
@miott , I added the changelog december.rst. Can you please this is correct.
Is there anything else I can do on my side ?

**Comment by miott** [comment]
> @miott , I added the changelog december.rst. Can you please this is correct. Is there anything else I can do on my side ?

I approved this pull request 5 days ago.  You need another approval.

File: connector/docs/changelog/2023/december.rst
Comment (ThomasJRyan) [inline connector/docs/changelog/2023/december.rst:1]: My apologies for this, but the changelog should go here https://github.com/CiscoTestAutomation/yang/tree/master/connector/docs/changelog/undistributed

We have a template in there for you to follow. Add it there and request a re-review from me

**Review by ThomasJRyan — CHANGES_REQUESTED** [review]

File: connector/docs/changelog/2023/december.rst
Comment (xvincent) [reply connector/docs/changelog/2023/december.rst:1]: @ThomasJRyan , I updated, added new changelog undistributed: fix_gnmi_reconnect.rst

**Review by ThomasJRyan — APPROVED** [review]

**Pull request merged**